test/jenkins/criu-iter.sh

```diff
@@ -5,4 +5,6 @@ set -e
 source `dirname $0`/criu-lib.sh
 prep
 mount_tmpfs_to_dump
-./test/zdtm.py run --all --keep-going --report report --parallel 4 --iter 3 -x 'maps04' || fail
+
+ITER_EXCLUDE="-x netns_lock -x maps04"
+./test/zdtm.py run --all --keep-going --report report --parallel 4 --iter 3 $ITER_EXCLUDE || fail
```

Comment:
We could use an infinite loop for the server to enable support for more than one iterations:
```diff
--- a/test/zdtm/static/netns_lock.hook
+++ b/test/zdtm/static/netns_lock.hook
@@ -35,30 +35,29 @@ if sys.argv[1] == "--post-start":
         srv.bind(("127.0.0.1", PORT))
         srv.listen(1)
 
-        # We should accept connections normally from
-        # pre-dump client, post-restore client and
-        # the pre-restore client using SOCCR_MARK
-        # The pre-restore client without SOCCR_MARK
-        # should fail with a timeout
-
-        # Accept pre-dump client
-        cln, addr = srv.accept()
-        cln.sendall(str.encode("--pre-dump"))
-        cln.close()
-
-        # Accept pre-restore client with SOCCR_MARK
-        srv.setsockopt(socket.SOL_SOCKET, SO_MARK, SOCCR_MARK)
-        cln, addr = srv.accept()
-        cln.sendall(str.encode("--pre-restore"))
-        cln.close()
-        srv.setsockopt(socket.SOL_SOCKET, SO_MARK, 0)
-
-        # Accept post-restore client
-        cln, addr = srv.accept()
-        cln.sendall(str.encode("--post-restore"))
-        cln.close()
-
-        srv.close()
+        while True:
+            # We should accept connections normally from
+            # pre-dump client, post-restore client and
+            # the pre-restore client using SOCCR_MARK
+            # The pre-restore client without SOCCR_MARK
+            # should fail with a timeout
+
+            # Accept pre-dump client
+            cln, addr = srv.accept()
+            cln.sendall(str.encode("--pre-dump"))
+            cln.close()
+
+            # Accept pre-restore client with SOCCR_MARK
+            srv.setsockopt(socket.SOL_SOCKET, SO_MARK, SOCCR_MARK)
+            cln, addr = srv.accept()
+            cln.sendall(str.encode("--pre-restore"))
+            cln.close()
+            srv.setsockopt(socket.SOL_SOCKET, SO_MARK, 0)
+
+            # Accept post-restore client
+            cln, addr = srv.accept()
+            cln.sendall(str.encode("--post-restore"))
+ 
```

Reply:
Great, but is it okay if the server is not closed? It will be freed eventually I think.

Reply:
zdtm would send [`SIGKILL`](https://github.com/checkpoint-restore/criu/blob/cbaa59976dbf9db618a7bbf3e9f175f40bba72bc/test/zdtm.py#L1362) to the server process.